docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,387 @@
+# Gluster-Block Provisioning End-To-End Design
+
+This design is the result of the work of several
+people, most notably Humble Devassy Chirammal (@humblec),
+who started an ealier draft and gave many of the original
+ideas, but also Steve Watt (@wattsteve),
+Vijay Bellur (@vbellur),
+Pranith Kumar Karampuri (@pranithk),
+and Prasanna Kumar Kalever (@pkalever).
+
+## Motivation
+
+While GlusterFS volumes can be used for RWO as well as RWX (and ROX) volumes,
+that support is not complete and problematic in some regards:
+
+1. RWO volumes should have protection from concurrent mounting.
+  (Locking to a host.) This is currently not enforced by the GlusterFS
+  mount plugin, hence the GlusterFS volumes may not meet the
+  expectations of the users. (This could be enforced in the GlusterFS
+  mount plugin, but there are more problems to solve.)
```

Comment:
@obnoxxx above seems really unclear to me. May be we need to put it in a way that `glusterFS` is a  shared FS or meant for many read/writers.